README.md

```diff
@@ -295,16 +398,380 @@ Here is an example:
+
+Note 2: This part of the example uses symbols as the hash key
+type. You can also use strings, which is discussed further on. The
+`"skill-1"`, `"skill-2"` and `"skill-3"` object member strings in the
+`skills` JSON object are converted to intern-ed symbols, i.e.,
+`'skill-1`, `'skill-2` and `'skill-3`. This means that you need to use
+`list-of symbol` to ensure that the skill reference strings in the
+`skill-refs` arrays are also converted into symbols to avoid a type
+mismatch when looking up elements with `gethash`.
+
+Using the following JSON object as an Emacs Lisp string:
+
+``` emacs-lisp
+(setq json-people "{
+  \"people\": [
+    {
+      \"personal\": {
+        \"address\": { },
+        \"name\": {
+          \"first\": \"John\",
+          \"last\": \"Johnson\"
+        }
+      },
+      \"job\": {
+        \"company\": \"Good Inc.\",
+        \"title\": \"CEO\",
+        \"location\": {
+          \"country\": \"Norway\",
+          \"city\": \"Oslo\"
+        }
+      },
+      \"skill-refs\": [
+        \"skill-1\",
+        \"skill-3\"
+      ]
+    },
+    {
+      \"personal\": {
+        \"address\": { },
+        \"name\": {
+          \"first\": \"Sammy\",
+          \"last\": \"Smith\"
+        }
+      },
+      \"job\": {
+        \"company\": \"Good Inc.\",
+        \"title\": \"VP, Marketing\",
+        \"location\": {
+          \"country\": \"USA\",
+          \"city\": \"Peoria\"
+        }
+      },
+      \"skill-refs\": [
+        \"skill-2\",
+        \"skill-3\"
+      ]
+    }
+  ],
+  \"skills\": {
+    \"skill-1\": {
+      \"name\": \"Programming\",
+      \"description\": \"Can code like Richard S or Linus T\"
+    },
+    \"skill-2\": {
+      \"name\": \"Design\",
+      \"description\": \"Can think out loud, structure code systems\"
+    },
+    \"skill-3\": {
+      \"name\": \"Communication\",
+      \"description\": \"Can explain complicated things to social scientists\"
+    }
+  }
+}")
+```
+
+The code below demonstrates how to use the `skills` hash table and
+symbols as keys:
+
+```emacs-lisp
+(let* ((people-data (jeison-read jeison-people json-people)))
+    ;; Do useful things with the data (in this case, just print):
+    (mapcar (lambda (person)
+              (princ (format "%s skills: %s\n"
+                             (oref person name)
+                             (mapconcat (lambda (the-skill-ref)
+                                          (let ((the-skill (gethash the-skill-ref
+                                                                    (oref people-data skills))))
+                                            (oref the-skill name)))
+                                        (oref person skill-refs)
+                                        ", "))))
+            (oref people-data people))
+    (pp (oref people-data skills)))
+```
+
+The output in the *ielm* buffer should be:
+
+```
+Johnson skills: Programming, Communication
+Smith skills: Design, Communication
```

Comment:
I suggest not to print the hash table and only keep these two lines. And it should also match the output style of the rest of the document, ie to be inline with the code via `;; =>` .